lesson3/README.md

```diff
@@ -0,0 +1,55 @@
+# Lesson 3 - Listviews, Adapters, Debugging
+
+## ListViews
+### Concepts Behind ListViews
+In order to display a scrollable list of something in Android, you will need to use a `ListView`.  A `ListView` is what you see when you scroll through a list of clickable things in an app, like your friends list in Snapchat or your list of chats in Messenger.
+
+Unfortunately, this is not as simple as just making a `ListView` and attaching an `ArrayList` to it.  You also need an `Adapter`.  There are three levels of displaying scrollable and clickable data in Android: the data source, the `Adapter`, and the `ListView`.
+
+- **Data Source**: This is the actual variable that stores all your data, usually an `ArrayList`.
+- **Adapter**: This interprets the data source and tells the `ListView` what each row of the list should look like.
+- **ListView**: This handles displaying the full list of rows and having the logic for any onClick listeners.
+
+In our Snapchat example, the data source would be an `ArrayList` of strings of our Snapchat friends' names.  The `Adapter` would handle displaying the name in a `TextView`, and also displaying the little icon to indicate whether you have an unread snap from them.  The `ListView` would handle displaying all your friends and making it scrollable when there's not enough room on the screen for the whole list.  It would also handle the logic for displaying your snap when you click on a friend with an unread snap, or replying when you double-click on a friend.
+
+For simple lists where you just want to display a list of strings in a `TextView`, you can use Google's built-in `ArrayAdapter`.  If you want to make a more complex display, such as including a picture or a button on each row, then you will need to create your own custom `Adapter`.  Just like you made a layout .xml file when creating the view for your fragment, you can make a layout xml file for your `Adapter` to position where each component goes in the row.  For the built-in `ArrayAdapter` you can pass in Google's built-in `simple_list_item_1` as your layout file, which fills the row with a `TextView`.
+
+### Code
+
+Now that you know the concept behind making a `ListView` work, read through [this guide](https://github.com/codepath/android_guides/wiki/Using-an-ArrayAdapter-with-ListView) to learn how to code it (you can skip the section about row view recycling if you want).

```

Comment:
I would either get rid of it or integrate your code with the Internet
stuff. Like you can mix your own explanations and stuff from the website.
And then I would link to the full doc

On Fri, Jul 15, 2016 at 4:29 PM Bill Wong <notifications@github.com> wrote:

> In lesson3/README.md
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/13#discussion_r71036205>
> :
>
> > +### Concepts Behind ListViews
> > +In order to display a scrollable list of something in Android, you will need to use a `ListView`.  A `ListView` is what you see when you scroll through a list of clickable things in an app, like your friends list in Snapchat or your list of chats in Messenger.
> > +
> > +Unfortunately, this is not as simple as just making a `ListView` and attaching an `ArrayList` to it.  You also need an `Adapter`.  There are three levels of displaying scrollable and clickable data in Android: the data source, the `Adapter`, and the `ListView`.
> > +
> > +- **Data Source**: This is the actual variable that stores all your data, usually an `ArrayList`.
> > +- **Adapter**: This interprets the data source and tells the `ListView` what each row of the list should look like.
> > +- **ListView**: This handles displaying the full list of rows and having the logic for any onClick listeners.
> > +
> > +In our Snapchat example, the data source would be an `ArrayList` of strings of our Snapchat friends' names.  The `Adapter` would handle displaying the name in a `TextView`, and also displaying the little icon to indicate whether you have an unread snap from them.  The `ListView` would handle displaying all your friends and making it scrollable when there's not enough room on the screen for the whole list.  It would also handle the logic for displaying your snap when you click on a friend with an unread snap, or replying when you double-click on a friend.
> > +
> > +For simple lists where you just want to display a list of strings in a `TextView`, you can use Google's built-in `ArrayAdapter`.  If you want to make a more complex display, such as including a picture or a button on each row, then you will need to create your own custom `Adapter`.  Just like you made a layout .xml file when creating the view for your fragment, you can make a layout xml file for your `Adapter` to position where each component goes in the row.  For the built-in `ArrayAdapter` you can pass in Google's built-in `simple_list_item_1` as your layout file, which fills the row with a `TextView`.
> > +
> > +### Code
> > +
> > +Now that you know the concept behind making a `ListView` work, read through [this guide](https://github.com/codepath/android_guides/wiki/Using-an-ArrayAdapter-with-ListView) to learn how to code it (you can skip the section about row view recycling if you want).
>
> Uhh if I copy-paste stuff in here, should I get rid of my sample code
> below where I make a friends list? Alternatively, I could give them three
> options: the quick guide, my copy-paste medium guide, and the full long
> guide via link. What do you think?
>
> —
> You are receiving this because you are on a team that was mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/13/files/5750eca8ed1e8f6aba956ee709175911086c1e97#r71036205>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIhRRbpdsILqzT57WPFU9RI-Z5qXtlDPks5qV-2egaJpZM4JJHgL>
> .
>